components/bl-AnimateOnScroll-component/src/index.js

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useMemo, useRef } from 'react';
+import { Aos } from './aos';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const DevicesMap = {
+  mobile: 'mobile',
+  tablet: 'tablet',
+  all   : true,
+  none  : false,
+};
+
+export default function AnimateOnScroll({ component, eventHandlers, pods }) {
+  const {
+    display, classList, style, animationType, easing, side,
+    duration, offset, disableFor, delay, mirror, once, anchor,
+  } = component;
+  const { onAnimationEnter, onAnimationOut } = eventHandlers;
+  const elementPod = pods['animationPod'];
+
+  const animationOnScrollRef = useRef(null);
+
+  const animationName = useMemo(() => {
+    return animationType + (side === 'none' ? '' : `-${ side }`);
+  }, [animationType, side]);
+
+  useEffect(() => {
+    Aos.init({ disable: DevicesMap[disableFor] });
```

Comment:
will it work if we modify the `disableFor ` property in runtime?

Reply:
no, that's why I deleted the logic handler and data binding

Reply:
can we specify the `disable` in HTML as `data-aos-disable`?

Reply:
no